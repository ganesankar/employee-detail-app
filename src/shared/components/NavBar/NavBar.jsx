import React, {useState} from 'react';
import {Link, NavLink} from 'react-router-dom'
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import {makeStyles, withStyles} from "@material-ui/core";
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import LockOpenIcon from '@material-ui/icons/LockOpen';

const useStyles = makeStyles(theme => ({
  link: {
    display: 'block',
    color: '#fff',
    textDecoration: 'none',
    marginRight: 4
  },
  list: {
    width: 'calc(100% - 106px)',
    display: 'flex',
    justifyContent: 'flex-end'
  },
  userProfile: {
    marginRight: theme.spacing(2)
  },
  icon: {
    fontSize: 21
  },
  container: {
    display: 'flex'
  },
  logoutButton: {
    display: 'flex',
    cursor: 'pointer'
  }
}));

const StyledAppBar = withStyles(theme => ({
  root: {
    backgroundColor: '#326da8'
  }
}))(AppBar);

const LoginButton = () => {
  const classes = useStyles();
  return (
    <React.Fragment>
      <React.Fragment>
        <NavLink to='/login' className={classes.link}>Login</NavLink>
        <LockOpenIcon className={classes.icon}/>
      </React.Fragment>
    </React.Fragment>
  );
};

const LogoutButton = (props) => {
  const classes = useStyles();
  const userName = localStorage.getItem('userLoggedIn') ?
    JSON.parse(localStorage.getItem('userLoggedIn'))['name'] : '';
  return (
    <div className={classes.container}>
      <div className={classes.userProfile}>
        <b>Welcome {userName}!</b>
      </div>
      <div className={classes.logoutButton}
           onClick={props.onClick}>
        <span className={classes.link}>Logout</span>
        <ExitToAppIcon />
      </div>
    </div>
  );
};

const NavBar = (props) => {
  const classes = useStyles();
  const [isLoggedIn, setIsLoggedIn] = useState(localStorage.getItem('login_token') !== null);

  const handleLogout = () => {
    setIsLoggedIn(false);
    props.handleLogout();
  };

  const getNavBarLinks = () => {
    if (isLoggedIn || localStorage.getItem('login_token')) {
      return (
        <LogoutButton onClick={handleLogout}/>
      )
    } else {
      return (
        <LoginButton />
      )
    }
  };

  return(
    <div>
      <StyledAppBar position="static">
        <Toolbar>
          <Typography variant="subtitle1" color="inherit">
            <Link to='/' className={classes.link}>Employee MS</Link>
          </Typography>
          <div className={classes.list}>
            {getNavBarLinks()}
          </div>
        </Toolbar>
      </StyledAppBar>
    </div>
  )
};

export default NavBar;