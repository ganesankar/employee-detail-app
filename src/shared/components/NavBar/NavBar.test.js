import React from 'react';
import NavBar from './NavBar';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

const onHandleLogout = () => {
  console.log('logging out');
};

it('deep snapshot renders correctly for logged in user', () => {
  const token = 'P@sst0ken';
  const userName = 'Alice';
  const tree = renderer.create(
    <BrowserRouter>
    <NavBar handleLogout={onHandleLogout}
                                       token={token}
                                       userName={userName}/>
    </BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
