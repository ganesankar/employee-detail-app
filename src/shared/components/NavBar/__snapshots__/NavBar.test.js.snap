// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deep snapshot renders correctly for logged in user 1`] = `
<div>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root WithStyles(ForwardRef(AppBar))-root-7 MuiAppBar-positionStatic MuiAppBar-colorPrimary"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit"
      >
        <a
          className="makeStyles-link-1"
          href="/"
          onClick={[Function]}
        >
          Employee MS
        </a>
      </h6>
      <div
        className="makeStyles-list-2"
      >
        <div
          className="makeStyles-container-5"
        >
          <div
            className="makeStyles-userProfile-3"
          >
            Welcome 
            <b>
              Alice
              !
            </b>
          </div>
          <div
            className="makeStyles-logoutButton-6"
            onClick={[Function]}
          >
            <span
              className="makeStyles-link-1"
            >
              Logout
            </span>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
